packages/aws-cdk-lib/aws-events/test/event-bus.grant.test.ts

```diff
@@ -0,0 +1,488 @@
+        expect(grant.success).toBeTruthy();
+      });
+
+      test('creates resource policy for organization principal with condition', () => {
+        // GIVEN
+        const organizationPrincipal = new iam.OrganizationPrincipal('o-12345abcdef');
+
+        // WHEN
+        const grant = eventBus.grantPutEventsTo(organizationPrincipal, defaultSid);
+
+        // THEN
+        assertOrganizationResourcePolicy(stack, 'o-12345abcdef');
+        assertOnlyResourcePolicy(stack, grant);
+        expect(grant.success).toBeTruthy();
+      });
+    });
+
+    describe('imported event bus scenarios', () => {
+      test('adds warning when granting to service principal on imported bus', () => {
+        // GIVEN
+        const importedEventBus = EventBus.fromEventBusName(stack, 'ImportedBus', 'external-bus');
+        const servicePrincipal = new iam.ServicePrincipal('states.amazonaws.com');
+
+        // WHEN
+        const grant = importedEventBus.grantPutEventsTo(servicePrincipal);
+
+        // THEN
+        assertNoResourcePolicy(stack);
+        expect(grant.success).toBeTruthy();
+        assertHasWarning(stack, '/Stack1/ImportedBus',
+          'Unable to add necessary permissions to imported target event bus: arn:\\${Token\\[AWS\\.Partition\\.[0-9]+\\]}:events:\\${Token\\[AWS\\.Region\\.[0-9]+\\]}:\\${Token\\[AWS\\.AccountId\\.[0-9]+\\]}:event-bus/external-bus \\[ack: @aws-cdk/aws-events:eventBusAddToResourcePolicy\\]',
+          false);
+      });
+
+      test('creates identity-based policy for imported event bus from ARN without warnings', () => {
+        // GIVEN
+        const importedEventBus = EventBus.fromEventBusArn(stack, 'ImportedBus', eventBus.eventBusArn);
+        const role = new iam.Role(stack, 'Role', {
+          assumedBy: new iam.ServicePrincipal('lambda.amazonaws.com'),
+        });
+
+        // WHEN
+        const grant = importedEventBus.grantPutEventsTo(role);
+
+        // THEN
+        assertIdentityBasedPolicy(stack, {
+          'Fn::GetAtt': ['EventBus7B8748AA', 'Arn'],
+        });
+        assertOnlyIdentityPolicy(stack, grant);
+        assertNoWarnings(stack);
+        expect(grant.success).toBeTruthy();
+      });
+
+      test('creates identity-based policy for imported event bus from name without warnings', () => {
+        // GIVEN
+        const importedEventBus = EventBus.fromEventBusName(stack, 'ImportedBus', eventBus.eventBusName);
+        const role = new iam.Role(stack, 'Role', {
+          assumedBy: new iam.ServicePrincipal('lambda.amazonaws.com'),
+        });
+
+        // WHEN
+        const grant = importedEventBus.grantPutEventsTo(role);
+
+        // THEN
+        assertIdentityBasedPolicy(stack, {
+          'Fn::Join': ['', [
+            'arn:',
+            { Ref: 'AWS::Partition' },
+            ':events:',
+            { Ref: 'AWS::Region' },
+            ':',
+            { Ref: 'AWS::AccountId' },
+            ':event-bus/',
+            { Ref: 'EventBus7B8748AA' },
```

Comment:
similar to the previous comment, I expect to see some normal string value, instead of ref intrinsic function.